lib/Doctrine/DBAL/Schema/DB2SchemaManager.php

```diff
@@ -25,7 +25,7 @@ class DB2SchemaManager extends AbstractSchemaManager
     public function listTableNames()
     {
         $sql  = $this->_platform->getListTablesSQL();
-        $sql .= ' AND CREATOR = UPPER(' . $this->_conn->quote($this->_conn->getUsername()) . ')';
+        $sql .= ' AND CREATOR = UPPER(' . $this->_conn->quote((string) $this->_conn->getUsername()) . ')';
```

Comment:
Without this explicit typecast, this was failing with a *"TypeError: Argument must be of the type string, null given"* when the connection username is null. I chose to just typecast to `string` and not do something like `AND CREATOR IS NULL` in this case, for 2 reasons:

- first of all because I don't know DB2 so I have no idea if this is a nullable column;
- and perhaps more importantly, because that was what `quote()` was doing so far: type-casting `null` to an empty string, so at least we're not changing the current behaviour.

It could be interesting to get someone with some DB2 knowledge to check that this behaviour is the proper one, though.

Reply:
Regardless of whether it's a nullable column, a condition like `WHERE CREATOR = UPPER(NULL)` will always yield `NULL`. I can check what's going on later, but this `(string)` casting doesn't look good.

Reply:
Still, it's the same as suppressing an error with the `@` operator. Most likely, the `WHERE CREATOR = ''` condition won't yield any rows. Therefore, what's the point in running this query at all?

Reply:
I honestly don't know, that's why I brought this to your attention in the first place. I just wanted to emphasize that my patch doesn't change the current behaviour.

Reply:
Note that the [comment right above](https://github.com/doctrine/dbal/blob/e6dcadb6f4d099d297527d81617d132e28f1713a/lib/Doctrine/DBAL/Schema/DB2SchemaManager.php#L22) says:

> Apparently creator is the schema not the user who created it:

followed by a broken link to ibm.com.

So it looks like this code does not do what was intended in the first place anyway?

Reply:
> I just wanted to emphasize that my patch doesn't change the current behaviour.

It doesn't but it violates the purpose of adding explicit types: instead of addressing the issue revealed by the type system, it hides it.

According to what I see by running `SELECT NAME, CREATOR FROM SYSIBM.SYSTABLES WHERE TYPE = 'T'`, `CREATOR` is never empty defaulting to `SYSIBM.SYSTABLES` but the username may be empty if some different authentication schema is used.

Similarly to what we do with MySQL,
https://github.com/doctrine/dbal/blob/a7cffda89e6154cf75502f250d0120523e73fac0/lib/Doctrine/DBAL/Platforms/MySqlPlatform.php#L392
we can default it to the [`CURRENT USER`](https://www.ibm.com/support/knowledgecenter/en/SSEPGG_9.7.0/com.ibm.db2.luw.sql.ref.doc/doc/r0011138.html) expression. Something like:
```php
if ($user !== null) {
    $sql = 'UPPER(' . $this->_conn->quote((string) $this->_conn->getUsername() . ')';
} else {
    $sql = 'CURRENT USER';
}
```

Reply:
I totally trust you on this one! Fixed in e401695.